demo/deploy-bookbuyer.sh

```diff
@@ -51,8 +51,8 @@ spec:
     spec:
       serviceAccountName: bookbuyer
       nodeSelector:
-        kubernetes.io/arch: amd64
-        kubernetes.io/os: linux
+        kubernetes.io/arch: $SYS_ARCH
+        kubernetes.io/os: $SYS_OS
```

Comment:
The demo images we build are currently only published for linux/amd64 and windows/amd64, so up until now the demo has had an implicit pre-req that the cluster has enough amd64 nodes to accommodate these pods. Do you know what it would take to spin up an amd64 kind cluster on your M1 Mac?

I can't remember the exact reason these were added, but I think it was to keep these pods from getting scheduled on Windows nodes, but these images have since been built and published for Windows so it seems less important now.

Since the demo is primarily for developers building images from source, I wonder if it makes sense to not define any nodeSelector or affinity by default so developers can generally do whatever they want. For advanced use cases like mixed windows/linux clusters or amd64/arm64 clusters where you want particular pods on particular nodes, these scripts could always be tweaked manually, but I worry trying to be smarter about it will just make things unnecessarily complicated.

Reply:
I'm kinda down for the idea to rip out the nodeSelector and drop affinities  -
I agree with the idea of "making it smarter will make things complicated", for demo and local env play testing I think simpler is better and we can add our own complexities on top of it when needed, but the average developer probably doesn't need it.